chain-signatures/node/src/indexer.rs

```diff
@@ -78,13 +81,59 @@ pub struct ContractSignRequest {
     pub key_version: u32,
 }
 
+#[derive(Debug, Clone)]
+pub struct Indexer {
+    latest_block_height: Arc<RwLock<LatestBlockHeight>>,
+    last_updated_timestamp: Arc<RwLock<Instant>>,
+}
+
+impl Indexer {
+    const BEHIND_THRESHOLD: Duration = Duration::from_secs(60);
+
+    fn new(latest_block_height: LatestBlockHeight) -> Self {
+        Self {
+            latest_block_height: Arc::new(RwLock::new(latest_block_height)),
+            last_updated_timestamp: Arc::new(RwLock::new(Instant::now())),
+        }
+    }
+
+    /// Get the latest block height from the chain.
+    pub async fn latest_block_height(&self) -> BlockHeight {
+        self.latest_block_height.read().await.block_height
+    }
+
+    /// Check whether the indexer is on track with the latest block height from the chain.
```

Comment:
Do we need both is_on_track and is behind?

Reply:
nope, but nice to have